src/webui/abstractwebapplication.cpp

```diff
@@ -386,3 +374,22 @@ const QStringMap AbstractWebApplication::CONTENT_TYPE_BY_EXT = {
     { "png", Http::CONTENT_TYPE_PNG },
     { "js", Http::CONTENT_TYPE_JS }
 };
+
+QStringMap AbstractWebApplication::parseCookie(const Http::Request &request) const
+{
+    // [rfc6265] 4.2.1. Syntax
+    QStringMap ret;
+    const QVector<QStringRef> cookies = request.headers.value(QLatin1String("cookie"))
+                                            .splitRef(';', QString::SkipEmptyParts);
+    for (const auto &cookie : cookies) {
+        const int idx = cookie.indexOf('=');
+        if (idx < 0)
+            continue;
+
+        const QString name = cookie.left(idx).trimmed().toString();
```

Comment:
before you approve this, I'm having constant crash at this exact line, gdb backtrace:
```
(gdb) bt
#0  0x00007ffff487b990 in __memcpy_ssse3 () at /usr/lib/libc.so.6
#1  0x00007ffff56e6467 in QString::QString(QChar const*, int) () at /usr/lib/libQt5Core.so.5
#2  0x00007ffff56f1a63 in QStringRef::toString() const () at /usr/lib/libQt5Core.so.5
#3  0x00000000005df80b in AbstractWebApplication::parseCookie(Http::Request const&) const (this=this@entry=0xfa7450, request=...) at ../../src/webui/abstractwebapplication.cpp:389
#4  0x00000000005e53ef in AbstractWebApplication::sessionInitialize() (this=this@entry=0xfa7450) at ../../src/webui/abstractwebapplication.cpp:153
#5  0x00000000005e5841 in AbstractWebApplication::processRequest(Http::Request const&, Http::Environment const&) (this=0xfa7450, request=..., env=...)
    at ../../src/webui/abstractwebapplication.cpp:116
#6  0x000000000050231c in Http::Connection::read() (this=0x187e830) at ../../src/base/http/connection.cpp:79
```
Why??

BTW, the crash goes away with `split()` in place of `splitRef()`...